src/levanter/infra/ray_tpu.py

```diff
@@ -284,27 +366,93 @@ def get_slice_info(self):
             num_tpus_per_host = 4
             num_hosts = num_cores // 8
         ip_address = socket.gethostbyname(socket.gethostname())
-
-        return SliceInfo(
+        self._slice_info = SliceInfo(
             slice_name=pod_name,
             num_hosts=num_hosts,
             num_tpus_per_host=num_tpus_per_host,
             ip_address=ip_address,
         )
+        self.scale_actor_pool(num_hosts)
+        return self._slice_info
 
+    def run_remote_fn(self, remote_fn: RemoteFunction, runtime_env: dict) -> list[ray.ObjectRef]:
+        """Run the remote function on this slice.
 
-@dataclass(frozen=True)
-class SliceResource:
-    """A collection of all resources associated with a single TPU slice."""
+        NOTE: This runs the remote function in a different task. It does not block on the remote function call.
+        NOTE: This returns a list of Ray futures. If calling this method on a remote Actor, you will get a future of a list of futures."""
+        actors = self.get_all_actors_in_pool()
+        if not self._slice_info or len(actors) < self._slice_info.num_hosts:
+            raise Exception("Insufficient host actors; call setup() before calling run_remote_fn()")
+        futures_of_futures: list[ray.ObjectRef] = [actor.run_remote_fn.remote(remote_fn, runtime_env) for actor in actors]
+        return [ray.get(future_of_future) for future_of_future in futures_of_futures]
 
-    actor: ActorHandle
-    """Actor handle for the SliceActor"""
+    def teardown(self):
+        self.drain_actor_pool()
+        self._slice_info = None
+
+
+@ray.remote
+class TPUHostActor:
+    """
+    Actor that manages a single TPU host.
+    """
+
+    def __init__(self, slice_name: str):
+        self._awaitable: Optional[ray.ObjectRef] = None
+        self._host_info: Optional[TPUHostInfo] = None
+        self._slice_name = slice_name
+
+    def healthy(self) -> bool:
+        return not self.is_being_preempted()
+
+    def is_being_preempted(self) -> bool:
+        from levanter.infra.tpus import get_current_tpu_is_preempted
+
+        return get_current_tpu_is_preempted()
+
+    def get_host_info(self) -> TPUHostInfo:
+        self._node_id = ray.get_runtime_context().get_node_id()
+        self._num_tpus = TPUAcceleratorManager.get_current_node_num_accelerators()
```

Comment:
we might still need my hack for v4 and v5p on some versions of ray, but maybe it's fine on >=2.47 and we should just require that